pkg/server/server_test.go

```diff
@@ -92,7 +91,14 @@ var _ = Describe("ApplicationsServer", func() {
 			BeforeEach(func() {
 				log = makeFakeLogr()
 				kubeClient = &kubefakes.FakeKube{}
-				appsSrv = NewApplicationsServer(&ApplicationConfig{App: app.New(nil, nil, nil, kubeClient, nil)})
+				appFactory := &apputilsfakes.FakeAppFactory{}
+				appFactory.GetAppServiceStub = func(ctx context.Context, name, namespace string) (*app.App, error) {
+					return app.New(nil, nil, nil, nil, nil, kubeClient, nil), nil
```

Comment:
Because the application server wouldn't know to create the mocked `AppService`. We configure it via the factory so that when it goes to create an `AppService` during a call, it creates the mock.